src/audienceManager.ts

```diff
@@ -0,0 +1,98 @@
+import { SDKLoggerApi } from './sdkRuntimeModels';
+import {
+    FetchUploader,
+    XHRUploader,
+    AsyncUploader,
+    fetchPayload
+} from './uploaders';
+import Audience from './audience';
+
+export interface IAudienceMembershipsServerResponse {
+    dt: 'cam';  // current audience memberships
+    ct: number; // timestamp
+    id: string;
+    audience_memberships: Audience[];
+}
+
+export interface IAudienceMemberships {
+    currentAudienceMemberships: Audience[];
+}
+
+export default class AudienceManager {
+    public url: string = '';
+    public userAudienceAPI: AsyncUploader;
+    public logger: SDKLoggerApi;
+
+    constructor(
+        userAudienceUrl: string,
+        apiKey: string,
+        logger: SDKLoggerApi,
+    ) {
+        this.logger = logger;
+        this.url = `https://${userAudienceUrl}${apiKey}/audience`;
+        this.userAudienceAPI = window.fetch
+            ? new FetchUploader(this.url)
+            : new XHRUploader(this.url);
+    }
+
+    public async sendGetUserAudienceRequest(mpid: string, callback: (userAudiences: IAudienceMemberships) => void) {
+        this.logger.verbose('Fetching user audiences from server');
+
+        const fetchPayload: fetchPayload = {
+            method: 'GET',
+            headers: {
+                Accept: '*/*',
+            },
+        };
+        const audienceURLWithMPID = `${this.url}?mpid=${mpid}`;
+        let userAudiencePromise: Response;
+
+        try {
+             userAudiencePromise = await this.userAudienceAPI.upload(
+                 fetchPayload,
+                 audienceURLWithMPID
+             );
+
+            if (
+                userAudiencePromise.status >= 200 &&
+                userAudiencePromise.status < 300
+            ) {
+                this.logger.verbose(`User Audiences successfully received`);
+
+                const userAudienceMembershipsServerResponse: IAudienceMembershipsServerResponse = await userAudiencePromise.json();
+                const parsedUserAudienceMemberships: IAudienceMemberships = {
+                    currentAudienceMemberships: userAudienceMembershipsServerResponse?.audience_memberships
+                }
+
+                try {
+                    callback(parsedUserAudienceMemberships);
+                } catch(e) {
+                    this.logger.error('Error invoking callback on user audience response.');
```

Comment:
You're doing a lot of custom error loggers in this function, but you're also nesting a try/catch block when it's unnecessary.

Each of these loggers should instead do a `throw new Error()` and then you can use your catch to pass it to our logger.

It would make things cleaner.